### PR TITLE
global sort: fix forget to close writer when reader has error

### LINKED ISSUE #51508: AWS global sort merge step OOM
Root cause is that in merge step, when reader side meets error, we forget to close the writer

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #51508

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > simple

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

File: br/pkg/lightning/backend/external/merge.go [hunk 1]
```diff
@@ -133,10 +145,5 @@ func mergeOverlappingFilesInternal(
 			return err
 		}
 	}
-	err = iter.Error()
-	if err != nil {
-		return err
-	}
-
-	return writer.Close(ctx)
+	return iter.Error()
```
Comment (Benjamin2037) [inline br/pkg/lightning/backend/external/merge.go:148]: Can add UT？

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**